pkg/prometheus/promcfg.go

```diff
@@ -2499,11 +2499,79 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	if len(sc.Spec.KubernetesSDConfigs) > 0 {
 		configs := make([][]yaml.MapItem, len(sc.Spec.KubernetesSDConfigs))
 		for i, config := range sc.Spec.KubernetesSDConfigs {
-			configs[i] = []yaml.MapItem{
-				{
-					Key:   "role",
-					Value: strings.ToLower(string(config.Role)),
-				},
+			assetStoreKey := fmt.Sprintf("scrapeconfig/%s/%s/kubernetessdconfig/%d", sc.GetNamespace(), sc.GetName(), i)
+			if config.APIServer != nil {
+				configs[i] = []yaml.MapItem{
+					{
+						Key:   "api_server",
+						Value: config.APIServer,
+					},
+				}
+			}
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "role",
+				Value: strings.ToLower(string(config.Role)),
+			})
+
+			configs[i] = cg.addBasicAuthToYaml(configs[i], assetStoreKey, store, config.BasicAuth)
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], fmt.Sprintf("scrapeconfig/auth/%s/%s/kubernetessdconfig/%d", sc.GetNamespace(), sc.GetName(), i), store, config.Authorization)
+			configs[i] = cg.addOAuth2ToYaml(configs[i], config.Oauth2, store.OAuth2Assets, assetStoreKey)
+
+			if config.ProxyConfig != nil {
```

Comment:
there's a lot of repetition with existing code. We need a utility function that translates a ProxyConfig field into Prometheus YAML config.

Reply:
sure, I can do that in follow up PR since ProxyConfig is used in a few SDConfigs as well